vfs/fs_utils.cc

```diff
@@ -537,6 +537,33 @@ dirName(const std::string & filename)
 }
 
 /****************************************************************************/
+/* UX FUNCTIONS                                                             */
+/****************************************************************************/
+
+void
+checkWritability(const std::string & url, const std::string & parameterName)
+{
+    // try to create output folder and write open a writer to make sure 
+    // we have permissions before
+    try {
+        Datacratic::makeUriDirectory(url);
```

Comment:
> the file will be created anyways when the procedure runs
Not if it fails before.   Do we have support to create a tmp file (that will automatically delete itself when the program exit)?  We probably don't have that for S3.  :-(